go/libraries/doltcore/sqle/tables.go

```diff
@@ -960,6 +965,66 @@ func (t *AlterableDoltTable) DropColumn(ctx *sql.Context, columnName string) err
 	return t.updateFromRoot(ctx, newRoot)
 }
 
+// dropColumnData drops values for the specified column from the underlying storage layer
```

Comment:
I could use some extra scrutiny here since this is my first time working with the storage layer.

While testing diffs, I noticed that when you drop a column, there was no diff reported, even though that operation logically modifies your data by removing that column. I wanted that change to show up in diff history, so as part of DropColumn, I added this code to update the stored data to remove the column.

Reply:
@andrew-wm-arthur We have talked about this off and on. You ok with dropping a column touching every row in storage?

Reply:
We've changed the implementation here a few times, people have different opinions. It makes sense that it's causes incorrect diff output here.

Paying the table scan cost here isn't strictly necessary, but this is the direction we're moving toward with the new format, so I think we should do it here too